src/renderer/dx/DxRenderer.cpp

```diff
@@ -1459,6 +1460,9 @@ CATCH_RETURN()
 [[nodiscard]] HRESULT DxEngine::PaintSelection(const SMALL_RECT rect) noexcept
 try
 {
+    // If a clip rectangle is in place from drawing the text layer, remove it here.
+    LOG_IF_FAILED(_customRenderer->EndClip(_drawingContext.get()));

```

Comment:
Are there other things that need this? Is it so important that we should add an explicit "I  am done with text" phase to the Render contract?

Reply:
It is important but I really didn't want to pipe an entire additional phase through...

Reply:
Are we sure we’ve got all the post-text EndClips covered then?

Reply:
1. I could put another "end text phase" through...

2. I could put a "end phase(enum of phases)" through so any engine who needs to clean up from a phase could do so on every phase.

3. I could just do this for now and potentially set someone up for a headache later...

4. I could slap EndClip in the top of all of them so they always clean up before the next phase reducing the chances someone wouldn't be aware of this when editing DxEngine in the future...


I think I like 2 the best, but open to further opinions.

Reply:
i like 2 the most. we don't need to fill out the enum of types right now, so we don't have to add every phase up front

Reply:
but also 3 is enticing

Reply:
I'm going to go with 3 because I usually don't build a framework for something until bitten in the ass at least 2-3 times. Risk/reward, etc.